### PR TITLE
Decrease isLive delay

### PR DESCRIPTION
**Pull request by gadyaari**

### CODE REVIEW

File: KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java [hunk 1]
```diff
@@ -127,8 +127,8 @@ public boolean removeReferrer(ILiveEntryReferrer obj) {
 		}
 
 		public LiveEntryCache(KalturaLiveEntry liveEntry, KalturaMediaServerIndex serverIndex, String applicationName) {
-			this(liveEntry);
-			register(serverIndex, applicationName);
+			// Check if this constructor is even relevant? Might be possible to delete?
```
Comment (kalturaguy) [inline KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java]: Why this is removed, it's not relevant for this spefici feature?

File: KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java [hunk 2]
```diff
@@ -127,8 +127,8 @@ public boolean removeReferrer(ILiveEntryReferrer obj) {
...
+			register();
```
Comment (kalturaguy) [inline KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java]: there is a difference between the target branch and your branch, you should resolve the conflics

File: KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java [hunk 3]
```diff
@@ -181,28 +181,43 @@ private void loadAssetParams() {
 			impersonateClient = null;
 		}
 
-		public synchronized void register(KalturaMediaServerIndex serverIndex, String appName) {
-			logger.debug("Register [" + liveEntry.id + "]");
-			if(registered)
+		public synchronized void setReadyForPlayback() {
+			// TO DO: Should we close this flag when UnRegister is called???
```
Comment (kalturaguy) [inline KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java]: don't leave TODO in pull request :)

File: KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java [hunk 4]
```diff
@@ -470,15 +488,38 @@ public void onEvent(IKalturaEvent event){
...
+	protected  void onStreamReadyForPlayback(String entryId, final KalturaMediaServerIndex serverIndex) {
+		logger.debug("onStreamReadyForPlayback [" + entryId + "]");
+
+		LiveEntryCache liveEntryCache = null;
+
+		synchronized (entries) {
+
+			if (entries.containsKey(entryId)) {
+				liveEntryCache = entries.get(entryId);
+				liveEntryCache.index = serverIndex;
```
Comment (kalturaguy) [inline KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java]: liveEntryCache.index = serverIndex;   do we really need it?

File: KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/listeners/LiveStreamEntry.java [hunk 5]
```diff
@@ -1017,4 +1054,70 @@ public void onHTTPSessionCreate(IHTTPStreamerSession httpStreamerSession){
...
+                    logger.info("Raising STREAM_READY_FOR_PLAYBACK event");
+                    raiseReadyForPlaybackEvent();
+                }
+            } catch (Exception err) {
+                logger.error(err);
+            }
+        }
+
+        private void raiseReadyForPlaybackEvent() {
+            IMediaStream mediaStream = this.packetizerCupertino.getAndSetStartStream(null);
```
Comment (kalturaguy) [inline KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/listeners/LiveStreamEntry.java]: please add sanity checks + log for null values

File: KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java (hunk 1 above)
Comment (gadyaari) [inline KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java]: You removed this when we were sitting together. But nonetheless, these two lines are not relevant as we don't call "register" from here and I'm pretty sure this constructor is not being used at all.

File: KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java (hunk 3 above)
Comment (gadyaari) [inline KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java]: My bad. I'll move them to a personal note

File: KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java (hunk 4 above)
Comment (gadyaari) [inline KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java]: Yes we do.
We're calling liveEntryCache.setReadyForPlayback(), and in this method we're calling setEntryMediaServer which is where we send isLive. setEntryMediaServer requires serverIndex as one of its parameters therefore we're saving it as a private instance in the calss liveEntryCache

File: KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/listeners/LiveStreamEntry.java (hunk 5 above)
Comment (gadyaari) [inline KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/listeners/LiveStreamEntry.java]: Not sure we need to. If the packetizerCupertino is null we will never create an instance of ChunkIdHandler (check the LiveStreamPacketizerListener2:onLiveStreamPacketizerCreate method where we check if the packetizer is instance of LiveStreamPacketizerCupertino)

**Pull request closed without merging**